lib/toolshed/http.ex

```diff
@@ -67,6 +67,92 @@ defmodule Toolshed.HTTP do
     end
   end
 
+  @doc """
+  Perform a HTTP GET request for the specified URL
+
+  By default, the results are printed or you can optionally choose to download
+  it to a specific locatiion on the file system.
+
+  Options:
+
+  * `:dest` - Location to write the response to. Defaults to printing to the terminal.
+  * `:verbose` - Display request and response headers. Disabled by default.
+  """
+  def httpget(url, options \\ []) do
+    check_inets()
+
+    url = url_defaults(url)
+    dest = Keyword.get(options, :dest, nil)
+    verbose = Keyword.get(options, :verbose, false)
+
+    stream =
+      if dest != nil do
+        to_charlist(dest)
+      else
+        :self
+      end
+
+    request_headers = [{'User-Agent', 'curl'}]
+
+    if verbose do
+      display_headers(request_headers, ">")
+    end
+
+    Task.async(fn ->
+      {:ok, _ref} =
+        :httpc.request(
+          :get,
+          {to_charlist(url), request_headers},
+          [],
+          sync: false,
+          stream: stream
+        )
+
+      handle_stream(verbose)
+    end)
+    |> Task.await()
+
+    IEx.dont_display_result()
+  end
+
+  defp url_defaults(url) do
+    case URI.parse(url) do
+      %URI{scheme: nil} -> "http://#{url}"
+      _ -> url
+    end
+  end
+
+  defp display_headers(headers, prefix) do
+    Enum.each(headers, fn {k, v} ->
+      IO.puts(:stderr, "#{prefix} #{k}: #{v}")
+    end)
+  end
+
+  defp handle_stream(verbose) do
+    receive do
+      {:http, {_ref, :stream_start, headers}} ->
+        if verbose do
+          display_headers(headers, "<")
+        end
+
+        handle_stream(verbose)
+
+      {:http, {_ref, :stream, body}} ->
+        IO.write(body)
+        handle_stream(verbose)
+
+      {:http, {_ref, :stream_end, _headers}} ->
+        IO.puts("")
```

Comment:
```suggestion
        IO.puts("")
```
I assume this isn't needed.

Reply:
From the diff I can't actually tell what changed, the lines look the same. I added this to ensure the prompt will start on a newline after printing the response.